Define distributive lattices and their ideals

ayberkt opened the pull request:
Preparation for Stone duality on spectral locales.

martinescardo commented:
Let's discuss this in person next week. I feel that, for the purposes of Stone duality, it would be better to define distributive lattices purely equationally. (And then maybe prove some things that are part of your current definition as theorems, if necessary.)

ayberkt commented:
@martinescardo I have now changed the definition to the one in Stone Spaces and implemented the things in the previous definition as consequences of this definition.

martinescardo commented on source/Locales/DistributiveLattice/Definition.lagda:
```diff
@@ -0,0 +1,340 @@
+
+\end{code}
+
+Projection for the carrier set.
```
Projection -> Notation.
